src/builder.ts

```diff
@@ -78,16 +77,26 @@ export default class Builder {
 		Bluebird.resolve(this.docker.buildImage(inputStream, buildOpts))
 		.then((res: NodeJS.ReadWriteStream) => {
 
-			const outputStream = res
+			let errored = false;
+			// Create a stream to write parsed output from the docker daemon
+			const outputStream = new Stream.PassThrough();
```

Comment:
`outputStream` should be created next to `inputStream` and made `setReadable` as early as possible (line 76)
The `sup` error handler should also be set as early as possible.
(Move everything that doesn't use `res` out of the `then` line 78)

`layers` shouldn't be an attribute of this class as it is not used anywhere outside of `createBuildStream`.